source/simulator/helper_functions.cc

```diff
@@ -909,8 +909,164 @@ namespace aspect
 
     return (material_model->get_model_dependence().viscosity != MaterialModel::NonlinearDependence::none);
   }
-}
 
+
+
+  template <int dim>
+  void Simulator<dim>::apply_limiter_to_dg_solutions (const AdvectionField &advection_field)
+  {
+    AssertThrow (dim == 2,
+                 ExcMessage ("Only apply bound preserving limiter to the 2 dimentional problem"));
+    const QGauss<1> quadrature_formula_1 (advection_field.is_temperature() ?
+                                          parameters.temperature_degree+1 :
+                                          parameters.composition_degree+1);
+    const QGaussLobatto<1> quadrature_formula_2(advection_field.is_temperature() ?
+                                                parameters.temperature_degree+1 :
+                                                parameters.composition_degree+1);
+    /**
```

Comment:
And you can remove the second * here. The reason is the following:
Normal C++ comments like this one are marked with /* ..... */. The double star is used for doxygen documentation of functions (like in our header files), and will make clear to the system that this description should appear in the online documentation of this function. It does not work for comments inside the implementation of a function though.